### PR TITLE
SpeedReader feature control via feature flag

### PR DESCRIPTION
**Pull request by AndriusA**
Resolves https://github.com/brave/brave-browser/issues/9749

Follow up on https://github.com/brave/brave-core/pull/5142

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [ ] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/speedreader/speedreader_browsertest.cc [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #include "brave/app/brave_command_ids.h"
 #include "brave/common/brave_paths.h"
 #include "brave/components/speedreader/speedreader_switches.h"
+#include "brave/components/speedreader/features.h"
```
Comment (iefremov) [inline browser/speedreader/speedreader_browsertest.cc:11]: sorting

Comment (bridiver) [reply browser/speedreader/speedreader_browsertest.cc:11]: make sure to run `npm run lint` or ci will fail anyway

Comment (iefremov) [reply browser/speedreader/speedreader_browsertest.cc:11]: i think linter doesn't catch this...

Comment (bridiver) [reply browser/speedreader/speedreader_browsertest.cc:10]: that's what @darkdh said too, but it used to catch these. I wonder what changed?

Comment (AndriusA) [reply browser/speedreader/speedreader_browsertest.cc:10]: github interface not showing correctly, but should be fixed now

File: browser/ui/views/toolbar/brave_toolbar_view.cc [hunk 2]
```diff
@@ -10,13 +10,12 @@
...
 #include "base/bind.h"
-#include "base/command_line.h"
+#include "base/feature_list.h"
 #include "brave/browser/profiles/profile_util.h"
 #include "brave/browser/ui/views/toolbar/bookmark_button.h"
 #include "brave/browser/ui/views/toolbar/speedreader_button.h"
 #include "brave/common/pref_names.h"
-#include "brave/components/speedreader/speedreader_pref_names.h"
-#include "brave/components/speedreader/speedreader_switches.h"
+#include "brave/components/speedreader/buildflags.h"
```
Comment (bridiver) [inline browser/ui/views/toolbar/brave_toolbar_view.cc:18]: is there a dep for this buildflag?

File: browser/ui/views/toolbar/brave_toolbar_view.cc [hunk 3]
```diff
@@ -26,6 +25,11 @@
 #include "components/bookmarks/common/bookmark_pref_names.h"
 #include "components/prefs/pref_service.h"
 
+#if BUILDFLAG(ENABLE_SPEEDREADER)
+#include "brave/components/speedreader/features.h"
```
Comment (bridiver) [inline browser/ui/views/toolbar/brave_toolbar_view.cc:29]: is there a dep for `//brave/components/speedreader`?

File: components/speedreader/speedreader_whitelist.h [hunk 4]
```diff
@@ -48,6 +52,16 @@ class SpeedreaderWhitelist : public brave_component_updater::BraveComponent {
   base::WeakPtrFactory<SpeedreaderWhitelist> weak_factory_{this};
 };
 
+using OnComponentReadyCallback =
```
Comment (bridiver) [inline components/speedreader/speedreader_whitelist.h:55]: what are these changes for?

Comment (AndriusA) [reply components/speedreader/speedreader_whitelist.h:55]: committed by accident, started to work on the other follow-up issues, will back this out

File: browser/ui/views/toolbar/brave_toolbar_view.cc (hunk 3 above)
Comment (AndriusA) [reply browser/ui/views/toolbar/brave_toolbar_view.cc:29]: in `brave/browser/ui/BUILD.gn`. This doesn't change deps from before, only adds buildflag condition that wasn't there for pref name

Comment (bridiver) [reply browser/ui/views/toolbar/brave_toolbar_view.cc:29]: oh right, this part was moved, but it does appear to be missing the buildflag dep that is required for https://github.com/brave/brave-core/pull/5529/files#diff-3d72382b1c4fba0ded2fc0100530ff54R18

Comment (AndriusA) [reply browser/ui/views/toolbar/brave_toolbar_view.cc:29]: should be fine now

**Pull request merged**